lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,668 @@
+		}
+
+
+		/**
+		 * Processes the `data-wp-interactive` directive.
+		 *
+		 * It adds the default store namespace defined in the directive value to the
+		 * stack so it's available for the nested interactivity elements.
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+		 * @param array                                     $context_stack   The reference to the context stack.
+		 * @param array                                     $namespace_stack The reference to the store namespace stack.
+		 */
+		private function data_wp_interactive_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+			// In closing tags, it removes the last namespace from the stack.
+			if ( $p->is_tag_closer() ) {
+				return array_pop( $namespace_stack );
+			}
+
+			// Tries to decode the `data-wp-interactive` attribute value.
+			$attribute_value = $p->get_attribute( 'data-wp-interactive' );
+			$decoded_json    = is_string( $attribute_value ) && ! empty( $attribute_value )
+			? json_decode( $attribute_value, true )
+			: null;
+
+			/*
+			 * Pushes the newly defined namespace or the current one if the
+			 * `data-wp-interactive` definition was invalid or does not contain a
+			 * namespace. It does so because the function pops out the current namespace
+			 * from the stack whenever it finds a `data-wp-interactive`'s closing tag,
+			 * independently of whether the previous `data-wp-interactive` definition
+			 * contained a valid namespace.
+			 */
+			$namespace_stack[] = isset( $decoded_json['namespace'] )
+			? $decoded_json['namespace']
+			: end( $namespace_stack );
+		}
+
+		/**
+		 * Processes the `data-wp-context` directive.
+		 *
+		 * It adds the context defined in the directive value to the stack so it's
+		 * available for the nested interactivity elements.
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+		 * @param array                                     $context_stack   The reference to the context stack.
+		 * @param array                                     $namespace_stack The reference to the store namespace stack.
+		 */
+		private function data_wp_context_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+			// In closing tags, it removes the last context from the stack.
+			if ( $p->is_tag_closer() ) {
+				return array_pop( $context_stack );
+			}
+
+			$attribute_value = $p->get_attribute( 'data-wp-context' );
+			$namespace_value = end( $namespace_stack );
+
+			// Separates the namespace from the context JSON object.
+			list( $namespace_value, $decoded_json ) = is_string( $attribute_value ) && ! empty( $attribute_value )
+			? $this->extract_directive_value( $attribute_value, $namespace_value )
+			: array( $namespace_value, null );
```

Comment:
nitpick: indent

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/58066/commits/6cee3ba0d25a79b4b4e902c0054cd0b008da7176.